### PR TITLE
Document new coding style guide

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Start documenting the new coding style rules and conventions. Most of here are from PSR-12 with a non-exhaustive list of our "custom" rules. After this, anyone can contribute adding to the customized rules section basing upon our enforced CS.

**Checklist:**
- [x] Securely signed commits

### CODE REVIEW

**Review by MGatner — CHANGES_REQUESTED** [review]
This is awesome! Thanks for being so thorough. One inline comment.

File: contributing/styleguide.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+own projects. Usage is discretionary within your projects but strictly enforceable within the framework.
+
+We follow the [PSR-12: Extended Coding Style][psr12] plus a set of our own
+styling conventions.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",
+"MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+_Portions of the following rules are from and attributed to [PSR-12][psr12]. Unless otherwise stated,_
+_portions therein not included herein SHOULD be read as if included herein._
```
[the remark this answers is not in the document]
Comment (sfadschm) [reply contributing/styleguide.md:17]: I guess it means that even if we do not copy all rules of PSR-12 to our styling guide documentation explicitly, such (uncopied) rules still apply.

[the remark this answers is not in the document]
Comment (paulbalandan) [reply contributing/styleguide.md:17]: 🤦‍♂️ Sorry, guys. The legalese took over on that part. And yes, @sfadschm  is right. 😅

**Comment by lonnieezell** [comment]
@paulbalandan is this ready to merge?

File: contributing/styleguide.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kenjis) [reply contributing/styleguide.md:17]: @paulbalandan 
How about this?
```
_Even if we do not copy all rules of [PSR-12][psr12] to this coding styling guide explicitly, such uncopied rules still apply._
```

**Review by kenjis — APPROVED** [review]

**Review by MGatner — CHANGES_REQUESTED** [review]
Could we get a link to the repo with a very general description?

**Comment by paulbalandan** [comment]
> Could we get a link to the repo with a very general description?

I think I did not follow what you want ?

**Comment by MGatner** [comment]
I would like some mention that the coding style is enforced by CS Fixer, with a link to the CodeIgniter/style-guide repo and a one-sentence introduction on what this is/does. I can suggest the change when I'm on desktop later if that would be helpful.

**Comment by kenjis** [comment]
@MGatner v4.1.4 was released. Why don't you merge this for now?

**Comment by MGatner** [comment]
Yes this needs to get in - I will make that addition today.

File: contributing/styleguide.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+We follow the [PSR-12: Extended Coding Style][psr12] plus a set of our own
+styling conventions.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",
+"MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+_Portions of the following rules are from and attributed to [PSR-12][psr12]. Even if we do not copy all the rules to this coding style guide explicitly, such uncopied rules SHALL still apply._
+
+[psr12]: https://www.php-fig.org/psr/psr-12/
+
```
Comment (MGatner) [inline contributing/styleguide.md:19]:
```suggestion
## Implementation

Our team uses [PHP CS Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) to apply coding standard fixes automatically. If you would like to leverage these tools yourself visit the [Official CodeIgniter Coding Standard](https://github.com/CodeIgniter/coding-standard) repository for details.

```

**Review by MGatner — CHANGES_REQUESTED** [review]
Feel free to emend, but this is the kind of addition I would like to see.

**Review by MGatner — APPROVED** [review]

**Pull request merged**